### ISSUE TITLE
Error when attempting to "Add Trigger" to Clip Generators (Details in description)

### ISSUE THREAD

**Issue by ap-Di0** [Zartar/Skyrim-Behavior-Editor-#4]
CRITICAL ERROR: ProjectAnimData::appendClipTriggerToAnimData():Failed to Set Data!
We advise you to contact author etc. 

Is there something specifically wrong with my workflow, or is this a current known issue?

**Comment by ap-Di0** [comment]
Just to add some context, i've been attempting to add a new attacking animation branch under CombatReadyStateBehavior in the GiantProject.hkx. 

The tool successfully added all state machines and state info nodes, but at the final step in the Clip Generator it errored out. 

Also of note is that the application also gave me an error when initially opening the GiantProject.hkx ("Warning: Potential Errors found", etc), DebugLog.txt is empty. Project file was opened and saved in previous SBE versions, also worth noting.

**Comment by Zartar** [comment]
Thanks for the report. I removed that functionality (appendClipTriggerToAnimData()) from clip generators  because it was unneccessary but clearly I forgot to remove it properly. Thank you for the detailed report. I'll fix it ASAP.

**Comment by Zartar** [comment]
Fixed. Check out the latest version. You can ignore most of the potential errors. I was very cautious with my data validity evaluation functions so some of them are ignorable. But if you're having problems it's worth checking out the debuglog (also should be fixed now) and ask me about them.

**Comment by ap-Di0** [comment]
Thanks! Is this fix included in 0.99991? It seems to be dated from 4 days ago.

**Comment by Zartar** [comment]
No, it's in 0.99.1, the third release down. I changed the versioning.

**Comment by ap-Di0** [comment]
Got it. Opened a vanilla GiantProject.hkx, it still gives me the error message at the beginning. It did generate a log this time around. It complains about lack of SSE, informs of load times and threw a complaint about certain nodes. 

> giantbehavior.hkx: hkbKeyframeBonesModifier: Ref: #143: KeyframeFullRagdoll: keyframeInfo is empty!

giantbehavior.hkx: hkbPoweredRagdollControlsModifier: Ref: #819: PoweredRagdollMatching: Null bones!

giantbehavior.hkx: hkbPoweredRagdollControlsModifier: Ref: #824: PoweredRagdoll No Matching: Null bones!

Have not tried to add event data at the clip generator yet as i'll have to rebuild from scratch following the error from last time. It seemed to have completely corrupted that project. 

Will report back. 

One thing i wanted to ask, does your tool generate the animationsingledatafile data or is it still necessary to write that stuff manually? I heard that was a necessary step to prevent random T-poses.

**Comment by ap-Di0** [comment]
Tool now adds events properly to clip generators. 
Encountered a similar error message when attempting to add a new Animation under "Animations" on the main page. 

> CRITICAL ERROR:AnimCacheAnimSetData:::removeAnimationFromCache():animation hash is invalid!!!!
etc

**Comment by Zartar** [comment]
Thanks again for the reports! I've fixed the add animation issue. Regarding the animation text files, "animationdatasinglefile.txt" contains animation translation and rotation data if defined (the other clip generator data is almost certainly ignored by the game engine). You can edit this data in the animations UI that you used to add the new animation. 

"animationsetdatasinglefile.txt" contains animation hashes used by the engine animation caching system. I've provided a crude UI to edit this data called "Animation Cache Files" UI on the far right of the project UI. When you save your project these files are written to and exporting the project should export these files too.

**Comment by ap-Di0** [comment]
Adding animations now works smoothly. Appreciate the awesome support. I hope these reports don't get annoying.

I managed to add the animation translate/rotate info, as well as add the caching data. Rebuilt the behavior tree states and state machines with clip + transitions. 

After replacing the master animationdatasingle.txt and animationsetdatasinglefile.txt, as well as the giantproject, giant behavior and adding the new animation, CK seems to crash upon trying to load the giant Nif. 

It would also appear projects opened and edited in previous versions of SBE have broken compatibility. 

When exporting project to Skyrim, the tool didn't export the animationdatasingle files. I noted that it also exported seemingly unrelated project files which had been previously opened but not saved. 

I looked at a custom project on Nexus, the Centaur creature project, and in their deployed files, they have animationdata and animationsetdata folders, which seem to include more granular data pertaining to every single animation referenced in the project. AnimationSetData folder also includes something called "bound data". 

Is this kind of data necessary for properly adding a behavior?

**Comment by ap-Di0** [comment]
![centaur-example](https://user-images.githubusercontent.com/24596201/42321227-e1b35bee-8060-11e8-988a-73bf1eb58dd1.jpg)

**Comment by Zartar** [comment]
These reports are extremely helpful, keep em coming! Back up your files regularly if your're having issues. I'll add a proper backup system soon. 

As far as I know the single file versions of the animation text files are the only ones that are necessary.

**Comment by ap-Di0** [comment]
There was an animationsetdatasinglefile and animationdatasinglefile generated in the local project folder correct. Thou exporting from within the editor didn't export those two files.

**Comment by ap-Di0** [comment]
Interesting. Unfortunately, every behavior edit i've tried to input into the game has caused it to crash. My edits largely follow vanilla trees to a tee. There also seems to be an issue right now where exporting to data folder exported a bunch of unrelated files which overwrote vanilla files. Random skeleton.nif and hkx files overwritten for seemingly no reason. It's caused a few issues trying to load vanilla assets. Curiously, those files coincide with hkx projects i've opened but never saved.

**Comment by Zartar** [comment]
Does the game crash before the main menu or on load? In the case of the former, remove the animation text files from the meshes folder and try loading the game. The tool cannot move any non-hkx files so I'm not sure what's going on with that.

**Comment by Zartar** [comment]
Sorry about the delayed response, I've been really ill for the last four days. I've discovered and fixed a major bug I introduced recently so make sure to use the latest version of the tool.

My work flow is pretty simple. Extract BSA, open project, edit, save project, export and test. I usually save and export individual behavior files to save time (no point in saving and exporting the whole project when you've just changed one file). You're a bit unlucky as I broke a few things with some of my careless recent updates but the next release should not have any such issues.
